### ISSUE TITLE
"failed scraping envoy metrics" when using gRPC proxyless

### ISSUE THREAD

**Issue by cfredri4** [istio/istio#38876]
### Bug Description

I have deployed some gRPC proxyless services [as described here](https://istio.io/latest/blog/2021/proxyless-grpc/). Both client and server can successfully communicate with each other.

However, as soon as the `inject.istio.io/templates: grpc-agent` annotation was added, the following error started to appear repeatedly in the `istio-proxy` container logs:

```
2022-05-12T13:42:06.948261Z	error	failed scraping envoy metrics: error scraping http://localhost:15090/stats/prometheus: Get "http://localhost:15090/stats/prometheus": dial tcp [::1]:15090: connect: connection refused
2022-05-12T13:42:09.649044Z	error	failed scraping envoy metrics: error scraping http://localhost:15090/stats/prometheus: Get "http://localhost:15090/stats/prometheus": dial tcp [::1]:15090: connect: connection refused
...
```

Log **with** `grpc-agent` annotation (there is no `istio-init` container injected into the pod; I assume this is as intended?):
* [istio-proxy log](https://gist.github.com/cfredri4/ee1c3aed05e8be834c7c9d8f2a8ec57a)

Logs **without** `grpc-agent` annotation:
* [istio-proxy log](https://gist.github.com/cfredri4/b1dfd0b22faf2b6ddaf6f2fb1188eedb)
* [istio-init log](https://gist.github.com/cfredri4/63a27529e45a3f110b991cd17c747151)

### Version

```prose
$ kubectl version --short
Client Version: v1.19.3
Server Version: v1.21.8

$ istioctl version
client version: 1.13.3
istiod version: 1.13.2
```

**Comment by zah-ra** [comment]
Did you find any solution for this issue? I have the same issue in my k8s cluster.

**Comment by cfredri4** [comment]
No solution found. Still waiting for input from Istio devs
This was found together with https://github.com/grpc/grpc-java/issues/9157. 
Perhaps @costinm who commented on that issue can take a look?

**Comment by Patrick0308** [comment]
It fixed by #38919 . It not be released yet.
